### ISSUE TITLE
Busy Box kernal panic

### ISSUE THREAD

**Issue by Ejmuse** [hakuna-m/wubiuefi#248]
When I launched ubuntu the first time it was fine, but when I restarted it brought me to a busy box open shell, when I typed exit it typed kernel failure. I tried a whole bunch of commands but all of them said file or directory not found. When I tried fsck it said fsct to util-Linux(version)

**Comment by hakuna-m** [comment]
> I tried a whole bunch of commands but all of them said file or directory not found.

Which ones ?

> When I tried fsck it said fsct to util-Linux(version)

fsck with which parameters ? What is the exact error message ? Is "fsct" a typo ?

Does the busy box show additional information like [here](https://github.com/hakuna-m/wubiuefi/issues/29#issuecomment-263599871).

**Comment by Ejmuse** [comment]
the error message appeared after I entered exit. It said kernel panic and then I couldn't type anything. All fsck commands I used came back with no directory or file found, I think windows might've deleted all the files from ubuntu for some reason. This happened after I changed somethings on ubuntu like changing the background and task bar.fsck commands I used were fsck -a /dev/sda1,fsck /dev/sda1 -y, fsck /dev/mapper/ubuntu--vg-root -y, and another command I used was ls /root/boot/grub. Like I said all of them came back with no file or directory found.

![
(https://cdn.discordapp.com/attachments/774720435321176064/775005965308264508/image0.jpg)
![](https://cdn.discordapp.com/attachments/774720435321176064/775011333804523550/image0.jpg)

**Comment by hakuna-m** [comment]
> This happened after I changed somethings on ubuntu like changing the background and task bar.

Maybe, the system wrote a new kernel and new boot files which do not work. Can you boot with an older kernel version ? see https://askubuntu.com/q/82140
 
> fsck commands I used were fsck -a /dev/sda1,fsck /dev/sda1 -y, fsck /dev/mapper/ubuntu--vg-root -y

I assume that Ubuntu on your "HP Pavilion x360" does not use device names like **/dev/sda1**.  Probably, the device name is **/dev/nvme0n1p1** instead of **/dev/sda1**.

But IMHO, it does not solve your issue if you correct **/dev/nvme0n1p1** with fsck. If we assume a typical Wubi layout, you see the reason:

Windows | Ubuntu | description
---|--|--- 
hidden | /dev/nvme0n1p1 | EFI System
hidden | /dev/nvme0n1p2 | Microsoft reserved
C: | /dev/nvme0n1p3 | Microsoft basic data
D: | /dev/nvme0n1p4 | Windows recovery environment
C:\ubuntu\disks\root.disk | [mount point]/ubuntu/disks/root.disk | Wubi disk

In the example above **/dev/nvme0n1p1* is the EFI partition and not the Wuibi disk. And also /dev/nvme0n1p3 is not the Wubi disk. It is Windows C: drive which you should  correct with Windows tools if it is really necessary.

If you want to correct the Wubi disk, it is necessary that the Windows drive with Ubuntu is already mounted.

So you can try to display existing mount points with command

`mount | grep /dev/`

**Comment by Ejmuse** [comment]
Do I do the command on windwos or Ubuntu,and also how do I get /dev/nvme0n1p1 and dev/nvme0n1p3 mounted on the wubi disk

**Comment by hakuna-m** [comment]
> Do I do the command on windwos or Ubuntu

on Ubuntu within BusyBox.

* display error messages:

`dmesg | grep -i error`

* display available devices

`blkid `

* display mounted devices

`mount | grep /dev/`

* mount Windows drive with Wubi disks, replace /dev/sda3 with the right device name reported by blkid

```
mkdir /windows
mount -t ntfs /dev/sda3 /windows
```

* check if Wubi disk is available, check filesystem on Wubi disk

```
ls -l /windows/ubuntu/disks/root.disk
fsck -fy /windows/ubuntu/disks/root.disk
```

* mount Wubi disk, replace /dev/loop0 with the device reported by losetup

```
mkdir /wubi
losetup -fv /windows/ubuntu/disks/root.disk
mount -t ext4 /dev/loop0 /wubi
```

* check if boot files exist

```
ls -l /wubi/boot/vmlinuz*
ls -l /wubi/boot/initrd*
```

* you can also store the output of the commands in file C:/ubuntu/log.txt e.g.

```
dmesg | grep -i error > /windows/ubuntu/log.txt 
blkid >> /windows/ubuntu/log.txt
mount >> /windows/ubuntu/log.txt
```
 
* unmount mounted Wubi disk and Windows drive

```
umount /dev/loop0
losetup -d /dev/loop0
umount /dev/sda3
```
* try again

`reboot `

**Comment by Ejmuse** [comment]
ill do everything you said and ill get back to you tonight

**Comment by Ejmuse** [comment]
I actually reinstalled ubuntu again,ill comment on this issue when it happens again,and ill tell you if all the commands worked

**Comment by hakuna-m** [comment]
If your current Ubuntu installation works, backup your file **C:\ubuntu\disks\root.disk** on Windows. If your installation does not work, you can just restore the file and get a working installation again. 

On Ubuntu in a terminal you can check the used devices and storage with

`df -h`

/host is the Windows drive where you installed Ubuntu.

`sudo losetup -a`

shows the used virtual disks with path. e.g.

/dev/loop0 which is on /host/ubuntu/disks/root.disk

**Comment by Ejmuse** [comment]
yes, the installation worked, how do I back the file up on windows, and whats the purpose of the backup? If the busy box open-shell opens again will I be able to restore to the backup?

**Comment by hakuna-m** [comment]
> If the busy box open-shell opens again will I be able to restore to the backup?

Yes, because you can still boot into Windows without reparing something on BusyBox.

> whats the purpose of the backup?

If you reinstall Ubuntu with Wubi, you loose your personal settings and data. If you have a backup, you can also restore those things. 

 > how do I back the file up on windows

see https://wiki.ubuntu.com/WubiGuide#How_can_I_make_a_backup_of_my_Wubi_install.3F

"The easiest method is to to simply copy the root.disk to an external medium. In the event that something goes wrong, copy the root.disk back to the Ubuntu folder on the Windows drive."

That method should also solve BusyBox issues. If you want to transfer your installation to a new computer, it is also possible to install Ubuntu with Wubiuefi and then you can copy your old  root.disk to the new installation.
